tests/Paket.Tests/Versioning/PlatformMatchingSpecs.fs

```diff
@@ -0,0 +1,200 @@
+﻿module Paket.PlatformMatchingSpecs
+
+open Paket
+open NUnit.Framework
+open FsUnit
+
+
+[<Test>]
+let ``Check that lists are updated``() =
+    // If this test fails it most likely means you need to
+    //  - Update the lists in KnownTargetProfiles
+    //  - Update base-lines (ie run Integration-Test-Suite) with update-base-lines set to true
+    //  - Review the diff + find a reviewer (on the PR) if you are unsure
+    //  - (In Extreme cases) Update this test.
+
+    let checkListEx (tagReader:obj -> int) (cases:Reflection.UnionCaseInfo[]) (l:'t list) =
+        let tags = l |> List.map tagReader
+        cases
+        |> Seq.forall (fun case ->
+            let foundCase = tags |> Seq.contains (case.Tag)
+            if not foundCase then
+                Assert.Fail (sprintf "Case '%s' was not found in KnownTargetProfiles.<type>Versions for '%s'" case.Name typeof<'t>.Name)
+            foundCase)
+        |> shouldEqual true
+        if l.Length <> cases.Length then
+            Assert.Fail (sprintf "KnownTargetProfiles.<list> doesnt't match number of cases for '%s'." typeof<'t>.Name)
+    let checkList (l:'t list) =
+        let tagReader = FSharp.Reflection.FSharpValue.PreComputeUnionTagReader(typeof<'t>)
+        let cases = FSharp.Reflection.FSharpType.GetUnionCases(typeof<'t>)
+        checkListEx tagReader cases l
+    checkList KnownTargetProfiles.DotNetFrameworkVersions
+    checkList KnownTargetProfiles.DotNetCoreAppVersions
+    checkList KnownTargetProfiles.DotNetStandardVersions
+    checkList KnownTargetProfiles.DotNetUnityVersions
+    checkList KnownTargetProfiles.MonoAndroidVersions
+    checkList KnownTargetProfiles.SilverlightVersions
+    checkList KnownTargetProfiles.UAPVersons
+    checkList KnownTargetProfiles.WindowsPhoneAppVersions
+    checkList KnownTargetProfiles.WindowsPhoneVersions
+    checkList KnownTargetProfiles.WindowsVersions
+
+    let reader = FSharp.Reflection.FSharpValue.PreComputeUnionTagReader(typeof<PortableProfileType>)
+    let unsupportedTag = reader(PortableProfileType.UnsupportedProfile [])
+    let cases =
+        FSharp.Reflection.FSharpType.GetUnionCases(typeof<PortableProfileType>)
+        |> Array.filter(fun case -> case.Tag <> unsupportedTag)
+    checkListEx reader cases KnownTargetProfiles.AllPortableProfiles
+
+[<Test>]
+let ``Can detect uap10``() =
+    let p = PlatformMatching.forceExtractPlatforms "uap10"
+    p.ToTargetProfile false |> shouldEqual (Some (SinglePlatform (FrameworkIdentifier.UAP UAPVersion.V10)))
+
+[<Test>]
+let ``Can detect uap10.1``() =
+    let p = PlatformMatching.forceExtractPlatforms "UAP10.1"
+    p.ToTargetProfile false |> shouldEqual (Some (SinglePlatform (FrameworkIdentifier.UAP UAPVersion.V10_1)))
```

Comment:
ah, sorry, I just saw the build from another build, and the log at the top, where it was also missing, but didn't actually look at the code.